### DISCUSSION TITLE
Overwriting blocks when extending blade

### DISCUSSION THREAD

**Discussion by TheGodlyLuzer** [laravel/framework#49126] [Ideas]
It would be amazing to have the possibility to override some blocks of code when extending a blade, similar to twig.

foo.blade.php
```php

<h1>title</h1>

@block("foo_main")
<h3>foo</h3>
@endblock
```

bar.blade.php
```php
@extends("foo")

@block("foo_main")
<h3>bar</h3>
@endblock
```

Would result in 

```php
<h1>title</h1>

<h3>bar</h3>
```

**Comment by DGINXREAL** [comment]
It's a good idea, like twig, if you extends an existing twig file, that allows you to override specified blocks

**Comment by rojtjo** [comment]
You can already do this by using the `@show` directive when defining a section:

`foo.blade.php`
```blade
<h1>title</h1>

@section('foo_main')
    <h3>foo</h3>
@show
```

`bar.blade.php`
```blade
@extends('foo')

@section('foo_main')
    <h3>bar</h3>
@endsection
```

Optionally keep parent content:

`bar.blade.php`
```blade
@extends('foo')

@section('foo_main')
    @parent

    <h3>bar</h3>
@endsection
```

More info here: https://laravel.com/docs/10.x/blade#layouts-using-template-inheritance

**Reply by DGINXREAL** [reply]
It's not working if you have multiple sections as children also. In my case the footer not showing anymore, and i just extends the file without any code

```blade
<h2>foobar</h2>
@section('content')
    <h3>Before Content</h3>
    @section('title')
        <h3>Title</h3>
    @show
    <h3>After Content</h3>
@show

@section('footer')
    <h3>Footer</h3>
@show

```

**Reply by rojtjo** [reply]
I've just tested a few combinations but for me it's working as intended. I feel like I've misunderstood what you meant here.

This is what I've tested: https://github.com/rojtjo/laravel-blade-sections-example

**Reply by DGINXREAL** [reply]
Sorry you have right, it was my mistake. Because i override `content` section.
